internal/install/discovery/os_windows_validator.go

```diff
@@ -0,0 +1,62 @@
+package discovery
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+)
+
+type OsWindowsValidator struct{}
+
+var (
+	WindowsVersionNoLongerSupported = "This version of Windows is no longer supported"
+	WindowsNoVersionMessage         = "Failed to identified a valid version of Windows"
+)
+
+func NewOsWindowsValidator() *OsWindowsValidator {
+	validator := OsWindowsValidator{}
+
+	return &validator
+}
+
+func (v *OsWindowsValidator) Execute(m *types.DiscoveryManifest) string {
+	if m.OS != "windows" {
+		return ""
+	}
+
+	versions := strings.Split(m.PlatformVersion, ".")
+
+	switch len(versions) {
+	case 0:
+		return WindowsNoVersionMessage
+	case 1:
+		major, err := strconv.Atoi(versions[0])
+		if err == nil {
+			return ensureMinimumVersion(major, 0)
+		}
+		return WindowsNoVersionMessage
+	default:
+		major, aerr := strconv.Atoi(versions[0])
+		if aerr == nil {
+			minor, ierr := strconv.Atoi(versions[1])
+			if ierr == nil {
+				return ensureMinimumVersion(major, minor)
+			}
+		}
+	}
+
+	return WindowsNoVersionMessage
+}
+
+func ensureMinimumVersion(major int, minor int) string {
```

Comment:
Instead of returning a string here, returning an error as we do in the `types` package I think would be more appropriate.  Then you could return `nil` for the case that it doesn't match.

Reply:
I'm not quite onboard with this, the main reason I find string easier to test with on the unit test side instead of adding several error.Error() calls to get back to the strings which makes the tests a lot harder to read.

In the end, a final error is raised after concatenating all validation messages into 1 string. 

Side note, the smaller validators (like OsValidator) are not meant to be used outside of the discovery package. Also, those validators are not erroring to do their work, they just signal back something potentially useful to the caller.

Reply:
Can you point out which test is simpler by returning a string here?  If you use the concrete types of an error, you don't have to check the `error.Error()` method, you can check the error type itself.

Reply:
Here is an example of using the `require` package in a unit test that compares two error types.  https://pkg.go.dev/github.com/stretchr/testify/require#Error

Reply:
Some impacted usages below.

It seems a few folks are lining up behind the usage of error versus string. I kind of wish go was having a nullable string type. To please the quorum, I'll go ahead and update, but makes me certainly prefer other languages. 

![image](https://user-images.githubusercontent.com/38018054/114599053-72f1ea80-9c47-11eb-8889-a2171912d65f.png)

![image](https://user-images.githubusercontent.com/38018054/114599102-869d5100-9c47-11eb-9eba-8c1d38a43d19.png)

Reply:
This might also be of interest for you.  https://pkg.go.dev/github.com/pkg/errors#hdr-Adding_context_to_an_error